test/testdata/infer/generics/all_bottom.rb

```diff
@@ -0,0 +1,30 @@
+# typed: true
+
+class Base1
+  extend T::Generic
+  Elem = type_member
+end
+
+class Base2
+  extend T::Generic
+  Elem = type_member
+end
+class Child < Base2
+  extend T::Generic
+  Elem = type_member
+end
+
+class Test
+  extend T::Sig
+
+  sig {params(x: Base1[T.untyped]).void}
+  def test1(x)
+    if x.is_a? Base2
+      T.absurd(x)
+    end
+
+    if x.is_a? Child
+      T.absurd(x)
```

Comment:
Would it also be the case that `puts` here would be unreachable? I'm curious what the revealed type of `x` is between these two `if`. It might be that we just don't have a type for `Base1 without subtypes of Base2`.

Reply:
Yes, that would also be unreachable. I used `T.absurd` because it captured the intent nicely, but I'd be happy to put in some other statements to show the errors being raised as well :)

Reply:
Revealing the type just shows `Base1[T.untyped]`; you're right that we don't have a way to capture negative information in types.